app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transferdetails/TransferDetailsScreen.kt

```diff
@@ -53,110 +58,152 @@ import java.util.Date
 @Composable
 fun TransferDetailsScreen(
     transferUuid: String,
+    direction: TransferDirection,
     navigateBack: (() -> Unit)?,
     transferDetailsViewModel: TransferDetailsViewModel = hiltViewModel<TransferDetailsViewModel>(),
 ) {
 
+    val getCheckedFiles = { transferDetailsViewModel.checkedFiles }
+    val clearCheckedFiles = { transferDetailsViewModel.checkedFiles.clear() }
+    val setFileCheckStatus: (String, Boolean) -> Unit = { fileUid, isChecked ->
+        transferDetailsViewModel.checkedFiles[fileUid] = isChecked
+    }
+
+    TransferDetailsScreen(transferUuid, direction, navigateBack, getCheckedFiles, clearCheckedFiles, setFileCheckStatus)
```

Comment:
We discussed that with @LunarX and @sirambd, we should do something like this:

```suggestion
    TransferDetailsScreen(
       transferUuid,
       direction,
       navigateBack,
       getCheckedFiles = { transferDetailsViewModel.checkedFiles },
       clearCheckedFiles = { transferDetailsViewModel.checkedFiles.clear() },
       setFileCheckStatus = { fileUid, isChecked ->
          transferDetailsViewModel.checkedFiles[fileUid] = isChecked 
       }
    )
```